test/replica-isolation/mzcompose.py

```diff
@@ -32,6 +34,135 @@
 ]
 
 
+class CompactionCheck:
+    # replica: a string describing the SQL accessible name of the replica. Example"cluster1.replica1"
+    # host: docker container name from which to check the log. Example: "computed_1_1"
+    def __init__(self, replica: str, host: str):
+        assert "." in replica
+        self.replica = replica
+        self.host = host
+        self.id: Optional[str] = None
+        self.satisfied = False
+
+    def find_id(self, c: Composition) -> None:
+        assert False
+
+    def print_error(self) -> None:
+        assert False
+
+    def check_log(self, c: Composition) -> None:
+        self.find_id(c)
+        assert self.id is not None
+        log: str = c.invoke("logs", self.host, capture=True).stdout
+        for line in [x for x in log.splitlines() if x.find("ClusterClient send=") > -1]:
+            if "AllowCompaction" in line:
+                if self.id in line:
+                    self.satisfied = True
+        if not self.satisfied:
+            self.print_error()
+
+    def replica_id(self, c: Composition) -> str:
+        cursor = c.sql_cursor()
+        (cluster, replica) = self.replica.split(".")
+        cursor.execute(
+            f"""
+                SELECT mz_cluster_replicas.id FROM mz_clusters, mz_cluster_replicas
+                WHERE cluster_id = mz_clusters.id AND mz_clusters.name = '{cluster}'
+                AND mz_cluster_replicas.name = '{replica}'""",
+        )
+        return cursor.fetchone()[0]
+
+    def cluster_id(self, c: Composition) -> str:
+        cursor = c.sql_cursor()
+        (cluster, replica) = self.replica.split(".")
+        cursor.execute(
+            f"SELECT id FROM mz_clusters WHERE mz_clusters.name = '{cluster}'",
+        )
+        return cursor.fetchone()[0]
+
+    @staticmethod
+    def _format_id(iid: str) -> str:
+        if iid[0] == "s":
```

Comment:
Idiomatic Python would be to use `iid.startswith("s")` instead. Also avoids the exception when `iid` is the empty string.